ts/defs.bzl

```diff
@@ -315,14 +315,24 @@ def ts_project(
         typecheck_target_name = "%s_typecheck" % name
         test_target_name = "%s_typecheck_test" % name
 
-        if type(transpiler) == "function" or type(transpiler) == "rule":
-            transpiler(
+        # To emulate resolveJsonModule in typescript, the transpiler should copy non-TS sources
+        # to the output tree. We do this just by checking if the transpiler reports that some srcs
+        # were unused (it didn't produce any outputs for them) and send those to our terminal
+        # js_library.
+        if type(transpiler) == "rule":
+            not_transpiled_srcs = transpiler(
```

Comment:
this one might not make sense, since a rule doesn't ever return anything

Reply:
Yea, that will always return None. You can only assume that all sources are consumed in this case. Perhaps mention that explicitly in the transpiler docstring above.